siad/daemon_test.go

```diff
@@ -21,19 +21,67 @@ func TestUnitProcessNetAddr(t *testing.T) {
 	}
 }
 
+// TestUnitProcessModules tests that processModules correctly processes modules
+// passed to the -M / --modules flag.
+func TestUnitProcessModules(t *testing.T) {
+	// Test valid modules.
+	testVals := []struct {
+		in  string
+		out string
+	}{
+		{"cghmrtwe", "cghmrtwe"},
+		{"CGHMRTWE", "cghmrtwe"},
+		{"c", "c"},
+		{"g", "g"},
+		{"h", "h"},
+		{"m", "m"},
+		{"r", "r"},
+		{"t", "t"},
+		{"w", "w"},
+		{"e", "e"},
+		{"C", "c"},
+		{"G", "g"},
+		{"H", "h"},
+		{"M", "m"},
+		{"R", "r"},
+		{"T", "t"},
+		{"W", "w"},
+		{"E", "e"},
+	}
+	for _, testVal := range testVals {
+		out, err := processModules(testVal.in)
+		if err != nil {
+			t.Error("processModules failed with error:", err)
+		}
+		if out != testVal.out {
+			t.Errorf("processModules returned incorrect modules: expected %s, got %s\n", testVal.out, out)
+		}
+	}
+
+	// Test invalid modules.
+	invalidModules := []string{"abdfijklnopqsuvxyz", "cghmrtwez", "cz", "z"}
+	for _, invalidModule := range invalidModules {
+		_, err := processModules(invalidModule)
+		if err == nil {
```

Comment:
I think it's better practice to check for a specific error. At the top of the file with the logic, declare an error:

```
var (
   ErrBadModuleString = errors.New("Unable to parse --modules flag, unrecognized or duplicate modules")
)
```

I still don't have a clean way for doing that with custom strings though.